### ISSUE TITLE
Non-deterministic test - OperatorPublishTest.testObserveOn

### ISSUE THREAD

**Issue by davidmoten** [ReactiveX/RxJava#2933]
Occurred in Travis build https://travis-ci.org/ReactiveX/RxJava/builds/61414126
```
rx.internal.operators.OperatorPublishTest > testObserveOn FAILED
    java.lang.AssertionError: expected:<1000> but was:<994>
        at org.junit.Assert.fail(Assert.java:93)
        at org.junit.Assert.failNotEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:128)
        at org.junit.Assert.assertEquals(Assert.java:472)
        at org.junit.Assert.assertEquals(Assert.java:456)
        at rx.internal.operators.OperatorPublishTest.testObserveOn(OperatorPublishTest.java:412)
```

**Comment by akarnokd** [comment]
Will be fixed with #2928: it's the classical case when I read some state before I read the terminal indicator which opens up a window where more values may come in between the two rendering the completion check incorrect.

**Comment by akarnokd** [comment]
Fixed via #2934.
